lib/jsonapi/utils/underscore.ex

```diff
@@ -20,10 +20,37 @@ defmodule JSONAPI.Utils.Underscore do
 
       iex> dash("-top--posts-")
       "-top--posts-"
+
+      iex> dash(%{"foo-bar" => "baz"})
+      %{"foo_bar" => "baz"}
+
+      iex> dash({"foo-bar", "dollar-sol"})
+      {"foo_bar", "dollar_sol"}
+
+      iex> dash(%{"f-b" => %{"a-d" => "z"}, "c-d" => "e"})
+      %{"f_b" => %{"a_d" => "z"}, "c_d" => "e"}
+
+      iex> dash(:"foo-bar")
+      :foo_bar
   """
   def dash(value) when is_binary(value) do
     String.replace(value, ~r/([a-zA-Z0-9])-([a-zA-Z0-9])/, "\\1_\\2")
   end
+  def dash(map) when is_map(map) do
+    Enum.into(map, %{}, &dash/1)
+  end
+  def dash({key, value}) do
+    {dash(key), dash(value)}
+  end
+  def dash(value) when is_atom(value) do
+    value
+    |> to_string()
+    |> dash()
+    |> String.to_atom()
+  end
+  def dash(value) do
+    value
+  end
```

Comment:
following up on the previous conversation, I think the swap of method names could come in another PR or in #137.  This looks great as is!